### ISSUE TITLE
More flexible train/validation

### ISSUE THREAD

**Issue by sammlapp** [kitzeslab/opensoundscape#370]
the train() function of PytorchModel assumes you want to run validation after every epoch. Might want to make this more flexible. Could also add flexibility with the metrics reported, and metrics used to choose the best model (as requested in #345)

**Comment by sammlapp** [comment]
In particular, printing F1, Precision, and Recall after running metrics_fn() is brittle and should be removed. Instead, the metrics function could have a verbose flag to print the calculated metrics.

**Comment by sammlapp** [comment]
I modified validation so that it will run the .eval() function, which the user can override to change (a) which metric is used to choose the best value, and (b) which metrics are calculated/saved. You can also now choose how often (in epochs) validation is performed.
